### ISSUE TITLE
Free Extended WPF Toolkit No Longer Permits Commercial Applications

### ISSUE THREAD

**Issue by AaronWrightAdm** [xceedsoftware/wpftoolkit#1557]
[A commit on Nov 26](https://github.com/xceedsoftware/wpftoolkit/commit/32c71eb1e7c83812bdb1dc99a7efee018321c6d5#diff-53e892b8b41cc4caece1cfd5ef21d6e7), part of the v3.7 release, replaced the MS-PL license with the Xceed Software, Inc Community License Agreement (for non-commercial use).  This blocks myself, and probably many others, from being able to upgrade to the latest public edition of the toolkit unless we pay hundreds if not thousands of dollars in licensing fees.  

Please revert to the MS-PL license.

**Comment by weitzhandler** [comment]
WOW!
Thanks for pointing that out.

**Comment by Dirkster99** [comment]
...but there are alternatives - open your eyes and you shall see:
- [Dirkster99/AvalonDock](https://github.com/Dirkster99/AvalonDock)
- [dotnetprojects/WpfExtendedToolkit](https://github.com/dotnetprojects/WpfExtendedToolkit)

I am having a feeling this issue entry won't have any effect as so many [discussions](https://github.com/xceedsoftware/wpftoolkit/pull/1523) that  [I've seen on similar subjects before](https://github.com/xceedsoftware/wpftoolkit/issues/1445). But rest assured your messages are [red, classified, and addressed](https://github.com/xceedsoftware/wpftoolkit/issues/1525#issuecomment-535131304) based on the demand.

**Comment by AaronWrightAdm** [comment]
Based on [this](https://github.com/xceedsoftware/wpftoolkit/pull/1523#issuecomment-535650782), it appears their way of getting "version parity" between paid and non-paid editions is to switch from monetizing updates for x months to monetizing commercial entities.  That's reflected in the new licensing.

I understand that alternatives exist, but I don't get to just swap things out and call it good.  I have to start by proving out that there is no intent to revert this licensing change.

**Comment by Suplanus** [comment]
All in all it is/was great from Xceed to do so in past, thank you.

As startup or small company its not that easy to start with Xceed now. I think others (like Syncfusion) going better ways with a small business license for free.

**Comment by XceedBoucherS** [comment]
Hi,

The Toolkit is still offered for free for your personal needs (or internal projects) with this community edition (packed with around 50 different controls). If you want to make business with the Toolkit you should be able to pay for the commercial license. Xceed also offers a small business subscription for less (perfect for startups or small companies : https://xceed.com/product/xceed-toolkit-plus-for-wpf/).
Thank you.

**Comment by AaronWrightAdm** [comment]
Thank you for confirming the intent of this licensing change, @XceedBoucherS .  I was only allowed to use this toolkit because the original licensing was compatible with commercial use for free.  I will be putting an item on our backlog to migrate away from use of this toolkit.

**Comment by CenturySpine** [comment]
Given that the release 3.6.0 and under are licensed under # Microsoft Public License (Ms-PL) is this possible to continue using these versions in commercial use ?

**Comment by XceedBoucherS** [comment]
Yes you can. But to get the latest updates, features and fixes, you will eventually need to migrate to a more recent version, which will be licensed for non-commercial projects.

Thank you.

**Comment by PennyDandy** [comment]
> The Toolkit is still offered for free for your **personal needs (or internal projects)** with this community edition

From the XCeed license :
> Licensee is **not authorized to:**
> 
> -sell or **license/sub-license/lease the resulting work to anyone** nor charge any amounts of money or exchange services for the said resulting work;

English nor legal are my first languages ; i gather that the new license also forbid the use wpftoolkit in all non-commercial open source applications. Can you confirm this interpretation of the license ? I was personally  evaluating the library for a non-commercial GPL software.

**Comment by XceedBoucherS** [comment]
Hi,
The important thing is that you (or any other developer or company) will NOT make money with the toolkit. License/sub-license/lease all results in making money : you can't do that.
But you can create an open source project which is non-commercial. This is kind of like the Toolkit on GitHub : an open source project free of use....but non-commercial.

**Comment by Suplanus** [comment]
And when is it a commercial project? The [COMMUNITY LICENSE AGREEMENT](https://github.com/xceedsoftware/wpftoolkit/blob/master/license.md) doesn't describe that... and I think it's all a matter of interpretation.

Have projects where someone buys me a beer, is it commercial?
Have projects that are funded by donations, is it commercial?

**Comment by XceedBoucherS** [comment]
A commercial project is when you use the toolkit in your project and you 
-sell your project 
-sell a license for users to use your project
-get any kind of money for your project
The Community license agreement says it : 
-"no re-selling, licensing or sub-licensing or any other rights of use".
-"sell or license/sub-license/lease the resulting work to anyone nor charge any amounts of money or exchange services for the said resulting work;"

1) Have projects where someone buys me a beer, is it commercial?
If the toolkit is included in the beer project, the company creating the beer project is selling it to make money : they need a **Commercial Toolkit license**. The Community Toolkit on GitHub can't be used for that. The company creating the beer project need to use the _Plus_ version of the Toolkit and pay the license from https://xceed.com/xceed-toolkit-plus-for-wpf/.
When someone is buying you this beer project, you can use it, it has been paid. But if you want to modify this beer project (which include the toolkit) and you want to sell it, you also need to buy the 
 **Commercial Toolkit license**. 

2) Have projects that are funded by donations, is it commercial?
If your project funded by donations includes the toolkit, you cannot sell this project with the community Toolkit on GitHub. By selling it, you make money : you need a **Commercial Toolkit license**, you  need to use the _Plus_ version of the Toolkit and pay the license from https://xceed.com/xceed-toolkit-plus-for-wpf/. But if you use this project for your own personal use, then the community Toolkit on GitHub is good enough.

**Comment by MarqueIV** [comment]
You shouldn't have changed it, dude!  I was a person who was committing PRs to fix your code before, but because of how you treated people on the free version, as less-than-important, not just less features, I'm out.  What you did was appalling.  You don't understand how open-source works.  

Yes, I get that you need to make money, and yes, I understand you have a paid version with more features.  However, most companies say you get 'X' for free, but if you pay, you *also* get Y, plus support.

In contrast, what you've been doing is saying people who pay get X and Y and if you don't pay, not only do you not get Y (understandable) but you get a much older, buggier version of X that's at least two versions behind.  

People like myself come along and try to support you, working on your codebase, pushing PRs to fix your issues... but you hold those fixes to the free version so you can give them to the paid users. Now you're taking from the free users too.

So where's my pay for fixing your code?  Code that you now charge for? Code I wouldn't have submitted if I had known you were going to block people from using it commercially, even in the free version?

Like I said in that other discussion... not cool.

At this point, I'm switching to another open-source version based on your 3.6 branch which is still usable for commercial products, and a branch who actually stays on top of bug fixes.  I really don't care if it deviates too far from what you have since what you've been sharing has been buggy anyway.

You really don't get Open Source, do you?

**Comment by MarqueIV** [comment]
You shouldn't have changed it, dude! I was a person who was committing PRs to fix your code before, but because of how you treated people on the free version, as less-than-important, not just less features, I'm out. What you did was appalling. You don't understand how open-source works.

Yes, I get that you need to make money, and yes, I understand you have a paid version with more features. However, most companies say you get 'X' for free, but if you pay, you also get Y, plus support.

In contrast, what you've been doing is saying people who pay get X and Y and if you don't pay, not only do you not get Y (understandable) but you get a much older, buggier version of X that's at least two versions behind.

People like myself come along and try to support you, working on your codebase, pushing PRs to fix your issues... but you hold those fixes to the free version so you can give them to the paid users. Now you're taking from the free users too.

So where's my pay for fixing your code? Code that you now charge for? Code I wouldn't have submitted if I had known you were going to block people from using it commercially, even in the free version?

Like I said in that other discussion... not cool.

At this point, I'm switching to another open-source version based on your 3.6 branch which is still usable for commercial products, and a branch who actually stays on top of bug fixes. I really don't care if it deviates too far from what you have since what you've been sharing has been buggy anyway and they, unlike you, do stay on top of their bugs because they actually care about their users, even the free ones.

You really don't get Open Source, do you?

**Comment by XceedBoucherS** [comment]
Hi,

First, as mentioned in another post, The Free Open source version of the Toolkit and the Plus Version of the Toolkit are coming to the same point. Currently, The OpenSource version is v3.7, while the Plus version is v3.8. The Free Open Source version v3.8 is about to be released in the next few days. So saying the Free version is always behind with PR behind held away from the community users, in a buggier version will not be an issue anymore.

Since the beginning of this project, you get 'X' for free, but if you pay, you also get Y, plus support. This was the case from v2.0 to v3.6 (about 7 years), but some people where making money from it without any contribution to the owners. A decision was taken so that if a company can make money from the Toolkit, they can at least pay the "small business license" at a small fee, and if the company do not make any money, than yes, they can continue to use the Free version as they wish. This is plan for now. So the OpenSource version is for non-commercial projects.

The Free OpenSource version will continue to be maintained and when bugs are found, we will continue to fix them.  The Toolkit is an important project for us and we will continue to make it progress in time.

**Comment by MarqueIV** [comment]
Well if that's the path you're taking, then why don't you just open-source *all* of the code, including the Plus version?  Put the restriction across all of it.  It's free for non-commercial, but paid if you use it commercially.  Having this double-whammy... not only do you *not* get all controls, but the controls you *do* get you have to now pay for, is not what most of us consider a good business decision.  Sure, you made that decision in what's right for you, but you still haven't addressed how we, in the community, committed to fixing your bugs with the understanding that it was for all to use for whatever.  You've essentially taken the work the community has done, and are saying 'Any future bugs, you now have to pay for.'  Again, this doesn't sit well with a lot of us. I personally was in the camp that if I needed the extra stuff in 'Plus', I'd have happily paid, but now since you're making people pay for the non-plus version, that just leaves too big a taste in my mouth, again, especially as someone who has either directly committed to, or via other developers helped commit to fixing bugs.  For instance, one bug around attributes I fixed *before* your license change, but you didn't merge it yet, meaning my fix is now going to go into a paid version.  That is not what I did it for. I did it for the community, not so you can make money off of my work without compensating me.  Like I said, not cool.

**Comment by XceedBoucherS** [comment]
Hi,
"Having this double-whammy... not only do you not get all controls, but the controls you do get you have to now pay for, is not what most of us consider a good business ". If you are working on a non-commercial project, you don't have to pay anything, nothing has changed for you with the new license. Again, if you make money with the Toolkit, we believe you can pay at least for the small business license (or see what is possible with the sales department).

We developed the whole Toolkit based on the community suggestions. Some members of the community have contributed with small fixes here and there, but the community didn't do it all. We prioritize the fixes and features based on the community needs and work on them. Usually, the community pull requests needs to be adapted/modified in the Toolkit because some tests couldn't pass. The Pull Requests from the community are never merged in directly from Nuget. We evaluates them, modify them and include the fix in the next release. Pull requests from the community is never going only in the Plus version only. Since the community version is what the community works with, they suggests fixes for the community version and this is where we put them. Again, it won't be long for the Community version to be at the same version than the Plus version is. Just be patient. 

But your comments are noted. Thank you

**Comment by RichardD2** [comment]
As a long-term subscriber to the Xceed Ultimate suite - at least 18 years now - I am becoming increasingly concerned with the direction Xceed are taking as a company.

We renewed our subscription last year, after receiving an email which felt like a threat. We were told that despite holding perpetual licenses for the components, if we didn't renew our subscription, we would not be allowed to modify any of our own code which uses an Xceed component.

Meanwhile, the commercial components have not been updated since January 2019. We have received precisely zero updates for our money. There is no roadmap. There are no news posts discussing any future updates. The support forums seem to be deserted.

Combined with this thread confirming that the "free" components developed with the help of the community will no longer be "free", I can only recommend to my company that we *do not* renew our subscription this year. Painful as it may be, we will be ripping all Xceed components out of our codebase, and replacing them with alternatives.

**Comment by MarqueIV** [comment]
Boucher, I understand your company feels that if we are using it in commercial software, we should pay, but what you are failing to understand are two things.

One, some of us who are creating commercial software are so far in the red because we are start ups with no external funding, that every single penny matters now.

More importantly, two, we chose to use your free components specifically because it was free for commercial use and are using it in software that we have been developing for the past four years which we still haven’t yet released because of the amount of effort a small firm needs when ‘going it alone’ for something like we’re doing.

We have supported your software specifically because it was free for commercial use. We have invested time and money in using your software specifically because it was free for commercial use. You changed your mind and have now screwed over the community who helped make your products what they are. You changed this *after* we have come to depend on your product.  Like a drug dealer, the first few are free, but once you’re dependent, you start charging.

If you had launched a product from the beginning saying that it was not free for commercial use, that would be fine because it would have given us valuable information needed to make our decisions.  As a result, yes, some of us might not have chosen it to begin with. But no one would be saying anything bad at all.

But that’s not what you did.  You did a bait-and-switch, which is frustrating enough, but just made that much more foul being something thing that was open source and actively supported by the community. Yes, you may have to ‘do the merging yourself’, but guess what... that’s what *every* owner of an open-source project has to do.  That’s table stakes.  That does not give you the right to suddenly start charging for something contributed to by the community.

Actually, I’m looking into seeing if by making this change Xceed has violated *their* responsibilities for open-source licensing by now charging for something that was contributed to by the open community under the guise of being free.  Maybe there’s nothing there, but I don’t think you realize just how much bad press and bad karma you’re putting out into the world with this change.

So that’s the part that you do not understand. Just like you have the right to charge for *your* work, when it’s work contributed by the community, it’s not just *your* work. It’s *our* work. And even if it’s a small, tiny part that the community committed, you have no way of knowing how important that was to get people to come to your project to make it successful.  Someone that makes spark plugs contributes a very small part of a car’s design, but the car ain’t gonna move without it!

My strong advice—all of ours—is to change the free version back to truly free, and stop neglecting it with changes.  That would be a way to turn this thing around and would show a good-faith effort to show that you not only heard us, but you’re working with us instead of against us.

Until that happens, both my personal not-yet-but-soon-for-profit company will stop using everything Xceed, paid or otherwise, but I will be making sure to share with all my clients how you arbitrarily change your licensing, which may even scare away paid users.  For all they know, they could pay for the product now, then you could change your license to a pay-per-install just because you feel like it.

In short, your company is proving it doesn’t stand by it’s word.  You need to fix that the same way you need to fix bugs in your code and stop sitting for months on PRs as well, just to charge others for the work later.

Anyway, at this point I’ve said all I’m going to say here.  I’m starting to feel my words are redundant.  I’m switching our code over to the still-open-and-free fork of this codebase and will be contributing our efforts there. Unless this license changes back, Xceed will no longer get our money, but also won’t get our support or recommendations.

**Comment by MarqueIV** [comment]
Boucher, I understand your company feels that if we are using it in commercial software, we should pay, but what you are failing to understand are two things.

One, some of us smaller one-or-two developer houses who are creating commercial software are so far in the red because we are start ups with no external funding, that every single penny matters now. Being commercial doesn't mean we are flush with cash.

More importantly, two, we chose to use your free components specifically *because* it was free for commercial use and are using it in software that we have been developing for the past four years which we still haven’t yet released because of the amount of effort a small firm needs when ‘going it alone’ for something like we’re doing. So for us, now we suddenly have to abandon parts of our codebase that we have depended on.

We have supported your software specifically because it was free for commercial use. We have invested time and money in using your software specifically because it was free for commercial use. You changed your mind and have now screwed over the community who helped make your products what they are. You changed this after we have come to depend on your product. Like a drug dealer, the first few are free, but once you’re dependent, you start charging.

If you had launched a product from the beginning saying that it was not free for commercial use, that would be fine. Open source projects do that kind of thing all the time.  That's fine because it would have given us valuable information needed to make our decisions *up front*. Yes, as a result, some of us might not have chosen it to begin with, but that would be their choice.  No one would have a problem with that. No one would fee betrayed.

But that’s not what you did. Whether you admit it or not, what you did was a bait-and-switch. You promised us something, we committed to using and supporting it, then you yanked it out from underneath us, which is frustrating enough, but just made that much more foul being something thing that was not only free, but something you made open source and which was actively supported by the community. Yes, you may have to ‘do the merging yourself’, but guess what... that’s what every owner of an open-source project has to do. That’s table stakes. That does not give you the right to suddenly start charging for something contributed to by the community.

Actually, I’m looking into seeing if by making this change Xceed has violated their legal responsibilities for open-source licensing by now charging for something that was contributed to by the open community under the guise of being free. I'm doing this because your change affects me directly and financially as a result.  Maybe there’s nothing there, but even if not, I don’t think you realize just how much bad press and bad karma you’re putting out into the world with this change as it relates to Xceed.

That’s the part we feel that you do not understand. I don't mean that in an insulting way, but rather an unvarnished critical way. Again while you absolutely have the right to charge for your work, when it’s work contributed by the community, I'm not sure that's legally allowed as it’s not just your work and at no time did any of us agree to it being sold for profit.

Remember, even if it’s a small, tiny part that the community committed, you have no way of knowing how important that was to get people to come to your project to make it successful. Someone that makes spark plugs contributes a very small part of a car’s design, but the car is not going to move without it!

My strong advice—all of ours here actually—is to change the free version back to truly free for all use, and stop neglecting it as compared to the plus version in regards to bug fixes.

I personally agree--as I'm sure others do too--that you can happily charge for all new components, or even enhancements to existing components (i.e. new styling or new properties x, y and z) and we are totally fine if you restrict them to the plus versions. But you need to restore what you have before, then be more attentive to fixes.

If you do that, we will be more than willing to keep contributing to the cause here and would be a great way to turn this thing around. A mea culpa. Companies, just like people, make mistakes. It's what you do afterwards. You have made a mistake. You are rubbing your own users the wrong way.  We feel betrayed.  Correcting it would show a good-faith effort to show that you not only heard us, but you’re working with us instead of against us.

Unfortunately, anything less than that and Xceed is no longer a company I can work with paid or otherwise, nor is it something I can actively support with any of my clients as I have been, happily, for the past ten years. After all, how do I know if we committed to a paid version, you wouldn't suddenly say 'I know you paid for it, but we feel since you've been *really* successful, we now get to charge you a per-install instead of a one-time fee.' You've changed licenses out from under us before.

Anyway, at this point I’ve said all I’m going to say here. I’m starting to feel my words are redundant. I’m switching our code over to the still-open-and-free fork of this codebase and will be contributing our efforts there. Unless this license changes back, Xceed will no longer get our money, but also won’t get our support or recommendations.

**Comment by XceedBoucherS** [comment]
Hi,
I'm sending your comments to the management. 
In the meantime, v3.8 could be released with the same license.
Thank you

**Comment by XceedBoucherS** [comment]
Hello,
2019 was an exceptional year at Xceed, where only 1 release of the Toolkit _Plus_ version was released. But this is not the plan for 2020. we are aiming for 2 or more versions throughout the year.

As for the Free _OpenSource_ Toolkit version, this product is still offered for free to any users not making any money from it, but also to any users which are developing a software with the Toolkit, but haven't made a sell yet. The only time the _Plus_ version (with a paying license) is necessary, is when you are selling your software. So if you are a small startup, developing a software with the free _OpenSource_ version of the Toolkit for years, it's all right. When you'll be ready to distribute your product, just make sure to buy the Toolkit _Plus_ version (with a commercial license).

This is the plan for now, but things could change even more in the favor of the Free _OpenSource_ users in a near future. Just stay tune !

**Comment by MarqueIV** [comment]
I appreciate the attempt to correct this here, but this still doesn't go far enough.  The free version should always be free, regardless.  The Plus version can always be paid, regardless.  What you said above, about it still being free 'during development' (i.e. before we sell software) still doesn't address the concerns I mentioned before, namingly a) we made a decision to use your software based on the understanding it would be free to use in commercial projects, and b) we committed to your open source project for the same reason.  The above does not address either of those issues.  This is still unsatisfactory from an open-source perspective.  Simply put, we believe legally you don't have the right to change the licensing for software that you allowed external contributions to under different licensing.  That is the initial feedback I have received from an OSS lawyer, which, if this continues, may become a more formal inquiry.

**Comment by bdachev** [comment]
@XceedBoucherS now that the library is under your custom license you should go trough code and remove any mentions to Microsoft Public License (Ms-PL) as it is a misleading.

**Comment by XceedBoucherS** [comment]
Hi bdachev, it is already done and will be included in the next release.
Thank you.

**Comment by efulton75** [comment]
I have a question about your commercial licensing that I think is best explained through a hypothetical:

Let's imagine that I work for Google (I don't) and I used your software to make an internal application that helps facilitate annual performance reviews (I didn't).  I don't sell this software or distribute this software outside of Google, all developers of it and users of it are internal.  But I'm certainly a commercial enterprise and you could argue that this software is used to make money because it helps us do performance reviews, which helps retain good talent and so we make more money.

Do I need to purchase your commercial licensing?  If so, for how many developers?  All internal Google developers?  The performance review code is open source internally so in theory anyone can change it, but realistically only a few do because everyone else is working on the website and gmail and stuff.

Curious how this affects my organization and whether we are able to upgrade.

**Comment by MarqueIV** [comment]
My advice, and this is as a contributor to this project… abandon this. Do not use this. Do not support this company.   Companies who change licensing and screw over their devoted users, especially after they were already abusing them by withholding updates, let alone essentially stealing for commercial purposes, work that was committed under the guise of open source from third parties such as ourselves, should no longer be supported. Neither my company, nor any company that I interface with will ever use Xceed again, and I will refuse to work with any companies who do.

Xceed changed its licensing and may have broken the law by putting a commercial license on code that was committed under the premise of being open source. Authors such as myself did not, nor would we ever authorize a company to profit from our work. From my lawyers initial response, they would have to retroactively go back and remove all third-party committed code, which they have not done, which means they are violating my copyright. From my lawyers initial response, they would have to retroactively go back and remove all third-party committed code, which they have not done, which means they are violating my copyright.  I have already filed paperwork in an attempt to get a judgment against Xceed. I will be updating here with my lawyer’s response shortly, possibly in the form of a cease and desist.

There is an open source fork of the Xceed work which you can use and that is still free and will remain free. We should be supporting that branch going forward and separating ourselves from Xceed.

https://github.com/dotnetprojects/WpfExtendedToolkit

**Comment by XceedBoucherS** [comment]
Hi efulton75,
Yes you can use the free version of the Toolkit for your internal needs. You are not making money from the Toolkit, therefor the commercial license do not apply.

As explained in another post, the free version and the commercial version are at the same point(or will be shortly if not already). There free version is no longer 3 versions behind the commercial version. The only differences are that the commercial version offer faster support and has more features and new controls.

As explained in another post, you don't have to use a commercial version if you are using the Toolkit for personal use, for educational purposes or while you are developing your commercial application. You only need to make sure you have a commercial version when you start to sell your application.

The Xceed Toolkit is owned by Xceed and Xceed offers a free version for the community. The community can propose fixes or patches. Xceed will reviews them, modify them and may include them in a future release by making sure it will not break any other controls and that it respects the Xceed coding standards.
Thank you

**Comment by mpoelzl** [comment]
Oh wow, I wasn't aware of this at all, thanks for pointing it out!

As a small developerI have been in similar situation before where supposedly 'free' licenses subsequently changed to being not free at all, not cool. Xceed, I appreciate the time and effort you have put into this. However if you take on an open source project then it should stay open source.

I am not a licensing expert but can you explain how you can take code from an open source project licensed under the MS-PL license and the license it under your own? Was there a deal made when you took on the project giving you rights to the source code?

**Comment by billn6** [comment]
Good morning @XceedBoucherS.  I'm restarting work on an indie commercial project from 2013 that uses the [AvalonDock version archived on CodePlex](https://archive.codeplex.com/?p=avalondock).  That version of the AvalonDoc code appears to be under BSD (referenced in the download zip file).  Will you confirm that license for that version and whether that version has restrictions on commercial use, please?

**Comment by Dirkster99** [comment]
This type of question/answer has [already been answered](https://github.com/xceedsoftware/wpftoolkit/issues/1557#issuecomment-572151528) based on a different release. You should take the time to read the thread despite its length - I think its well worth it.

**Comment by billn6** [comment]
Thanks for the link - it was informative and useful.

**Comment by alexhumphrey** [comment]
@XceedBoucherS I'm a bit confused on using the toolkit on an internal-use-only application within a business. You're quite clear above that you have to be making money from selling the software containing the toolkit, but the licence says "Licensee is not authorised to ... deploy the Software for/in a commercial environment;" - This would seem to outlaw internal use - is that correct? 
Thanks.

**Comment by XceedBoucherS** [comment]
Your assumption is correct however, we will soon announce some changes to our license that could be beneficial for you, keep an eye open the announcement should be coming soon. 
Until then, you can use this version of the Toolkit.
Thank you.

**Comment by grimreaper38uk** [comment]
I'm a NON-commercial developer.  I used to be professional developer, but now it's just a hobby.  I've read all the information I can find about this change in licensing.  I've just updated my Nuget source to the new v4.0.0.
My projects now fail to compile due to the lack of license.
I have only one question, after searching through all issues, Wiki pages, read me files, help files and Google results - how the f u c k do I get a non-commercial license to use the 'free' software?!?!

**Comment by XceedBoucherS** [comment]
Hi, grimreaper38uk,
Please try with the new v4.0.1, available on Nuget and GitHub.
There was a mismatch when v4.0.0 was uploaded.
Thank you for reporting this.

**Comment by adospace** [comment]
I'm the original author of AvalonDock (from back 2007), I spent endless hours of my personal time to allow the community of WPF opensource users a valid alternative to all the other commercial docking libraries. 
Sorry guys, I'm really sad now to see that AvalonDock is no more an open-source project, hope Xceed re-think this choice in the future.

**Comment by Dirkster99** [comment]
@adospace 
> I'm the original author of AvalonDock (from back 2007), I spent endless hours of my personal time to allow the community of WPF opensource users a valid alternative to all the other commercial docking libraries.
> Sorry guys, I'm really sad now to see that AvalonDock is no more an open-source project, hope Xceed re-think this choice in the future.

I am the original author of the Codeproject Tutorial which I think I've created about 2010 and I felt the same way as you do a few years ago, which is why I've started an [alternative branch](https://github.com/Dirkster99/AvalonDock) that is closing on *500* Stars and *100,000* nuget downloads :-)

These Stars and download figures would not be happening without your contribution - so don't be sad, because your project is alive and doing well - it just moved on to somewhere else :-)

**Comment by adospace** [comment]
@Dirkster99 thanks so much for your efforts, you have my star too!

**Comment by MarqueIV** [comment]
@adospace, If you are the original author and your original license said the code had to remain open source, then Xceed is not allowed to re-define It to be closed-source.

Additionally, there’s nothing stopping you or anybody else from forking the code at the last point it was still considered open source even by them, and just moving forward with that branch.

The bottom line is Xceed really burned their bridges  here. Not only are they now banned from being used in any future product we do, but all existing code is also being phased out/re-factored away so they are completely extricated from our codebase. For legal reasons, I can’t say the company’s name, but let’s just say they lost a very big ally.

**Comment by mdn1** [comment]
> 
> 
> I have a question about your commercial licensing that I think is best explained through a hypothetical:
> 
> Let's imagine that I work for Google (I don't) and I used your software to make an internal application that helps facilitate annual performance reviews (I didn't). I don't sell this software or distribute this software outside of Google, all developers of it and users of it are internal. But I'm certainly a commercial enterprise and you could argue that this software is used to make money because it helps us do performance reviews, which helps retain good talent and so we make more money.
> 
> Do I need to purchase your commercial licensing? If so, for how many developers? All internal Google developers? The performance review code is open source internally so in theory anyone can change it, but realistically only a few do because everyone else is working on the website and gmail and stuff.
> 
> Curious how this affects my organization and whether we are able to upgrade.

@XceedBoucherS is there any answer to this question?

**Comment by XceedBoucherS** [comment]
Hi mdn1,
It is used in a commercial context so yes, you will be required to have 1 license per developer using the product.
If you want to talk about it, I invite you to contact sales@xceed.com.
Thank you.

**Comment by XceedBoucherS** [comment]
Hi @Maximilian-Renner ,
Don`t bother about the attached license.txt with that version of Toolkit NuGet v3.8.2 is MS-PL.
Thank you

**Comment by Rumi1604** [comment]
I see below as license information on the versions
3.6.0 and before : MS-PL
3.7.0 : Xceed License
3.8.0 : Xceed License
3.8.1 : Xceed License 
3.8.2 : MS-PL -> Does this mean 3.7,3.8,3.8.1 as also falling under MS-PL as the license info on 4.0.0 says "Starting at v4.0.0, this free and open source toolkit is provided under the Xceed Community License agreement(for non-commercial use). Released June 26, 2020."
Its tricky as the license on those 3.7,3.8,3.8.1 versions remains xceed community.

**Comment by XceedBoucherS** [comment]
Hi @Rumi1604 ,
3.6.0 and before are MS-PL.
3.7.0, 3.8.0 and 3.8.1 were released with the Xceed License.
Then we decided to offer one last version under MS-PL : v3.8.2.
Then, starting at v4.0, all the following versions use the Xceed License (for non-commercial use).

So if you want to use an MS-PL version, use the latest one with that license(v3.8.2) and if you want an up to date version, use the v4.0+ versions (under the Xceed license).

Thank you and sorry for the confusion. That period was when the new license was created.

**Comment by park-bitset** [comment]
It's a surprise to see people down voting @XceedBoucherS 's comment.
They want to use a good program for free and want to make money out of it for free too?
That's just not how things work.
